src/main/java/com/yelp/nrtsearch/server/luceneserver/field/TermQueryableIndexableFieldDef.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.grpc.TermInSetQuery;
+import com.yelp.nrtsearch.server.grpc.TermQuery;
+import com.yelp.nrtsearch.server.luceneserver.field.properties.TermQueryable;
+import org.apache.lucene.search.Query;
+
+public abstract class TermQueryableIndexableFieldDef extends IndexableFieldDef
+    implements TermQueryable {
```

Comment:
I am not sure what being indexable adds to this class. Perhaps this could be added to the TermQueryable interface, with a default method implementations provided.

Reply:
Originally all the fielddefs (e.g. `BooleanFieldDef`, `IntFieldDef`, etc.) were all `<FieldDef> extends IndexableFieldDef implements TermQueryable, <other_interfaces>`. To be able to keep this structure while providing default method implementations / validation, I created this class. Is there a better way to do this?

Reply:
Would this be doable using interface default methods in TermQueryable, such as https://www.tutorialspoint.com/java8/java8_default_methods.htm ?

Reply:
Ah, didn't know default methods were possible in Java interfaces. This should work, will make the changes.